### DISCUSSION TITLE
How to interface with Gen2-API of Shelly Plus 1PM?

### DISCUSSION THREAD

**Discussion by deboiph** [arachnetech/homebridge-mqttthing#511] [Q&A] [answered]
How to configure the getOn topic for a Shelly Plus 1 PM through its [Gen2-API](https://shelly-api-docs.shelly.cloud/gen2/)?
The response to a getStatus request seems to be asynchronous (reply to another topic).
Here is what I have so far:

            "topics": {
                "getOnline": "shellyplus1pm-xxxxxxxxxxxx/online",
                "getOn": "shellyplus1pm-xxxxxxxxxxxx/rpc",    <= do not work
                "setOn": {
                    "topic": "shellyplus1pm-xxxxxxxxxxxx/rpc",
                    "apply": "return JSON.stringify({id:1, src:'shellyplus1pm-xxxxxxxxxxxx', method:'Switch.Toggle', params:{id:0}})"
                }

The getOline and setOn topics are working fine.

Thanks,
pHiL

**Comment by phillipse01** [comment, marked as answer]
Taken from the second example at https://github.com/arachnetech/homebridge-mqttthing/wiki/Tested-Configurations#shelly-plus-1 the following works:
`"getOn": {
            "topic": "shellyplus1pm-xxxxxxxxxxxx/events/rpc",
            "apply": "if (message.includes('output')) return JSON.parse(message).params[\"switch:0\"].output;"
        }`

**Reply by deboiph** [reply]
In the meantime, I finally succeeded and my configuration is very close to the one of the second example you mentioned. Thanks anyway!
